dd-sdk-android-androidx-fragment/src/main/kotlin/com/datadog/android/androidx/fragment/FragmentViewTrackingStrategy.kt

```diff
@@ -14,23 +15,47 @@ import com.datadog.android.rum.ViewTrackingStrategy
  *
  * **Note**: This version of the [FragmentViewTrackingStrategy] is compatible with
  * the AndroidX Compat Library.
+ * @param trackArguments whether we track Fragment arguments
  */
-class FragmentViewTrackingStrategy : ActivityLifecycleTrackingStrategy(), ViewTrackingStrategy {
+class FragmentViewTrackingStrategy(private val trackArguments: Boolean) :
+    ActivityLifecycleTrackingStrategy(), ViewTrackingStrategy {
 
     // region ActivityLifecycleTrackingStrategy
 
+    private val compatLifecycleCallbacks: FragmentManager.FragmentLifecycleCallbacks
+            by lazy {
+                CompatFragmentLifecycleCallbacks {
+                    if (trackArguments) {
+                        it.arguments.asRumAttributes()
+                    } else {
+                        emptyMap()
+                    }
+                }
+            }
+    private val oreoLifecycleCallbacks: android.app.FragmentManager.FragmentLifecycleCallbacks?
+            by lazy {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    OreoFragmentLifecycleCallbacks {
+                        if (trackArguments) it.arguments.asRumAttributes() else emptyMap()
+                    }
+                } else {
+                    null
+                }
+            }
+
```

Comment:
I did not wanted to create a new interface...but indeed it looks much cleaner this way.

Reply:
Actually this idea with the interface it's good but it doesn't actually work like that. In your lazy builder you do not have access to the activity instance. Also I need to do use a generic for the interface to avoid doing checks int he register/unregister methods to see if the activity is a FragmentActivity or not. This will require some tweaks. Will give it a try to see if I can get something better.